src/radical/pilot/agent/lm/jsrun.py

```diff
@@ -66,25 +70,34 @@ def _create_resource_set_file(self, slots, uid, sandbox):
 
         uid     : unit ID (string)
         sandbox : unit sandbox (string)
+        mpi     : MPI or not (bool, default: False)
 
         """
 
-        rs_id  = 0
-        rs_str = ''
-
+        rs_str = 'cpu_index_using: physical\n'
+        rank = 0
         for node in slots['nodes']:
-
-            cores = ' '.join([str(core_set[0]) for core_set
-                                               in  node['core_map']])
-            gpus  = ' '.join([str(gpu_set[0])  for gpu_set
-                                               in  node['gpu_map']])
-
-            rs_str           += 'RS %d: {'  % rs_id
-            rs_str           += ' host: %d' % node['uid']
-            if cores: rs_str += ' cpu: %s'  % cores
-            if gpus : rs_str += ' gpu: %s'  % gpus
-            rs_str           += ' }\n'
-            rs_id            += 1
+            if node['gpu_map']: 
+                node_map = zip(node['core_map'], node['gpu_map'])
+            else:
+                node_map = zip(node['core_map'], *node['gpu_map'])
```

Comment:
What does this do?

Reply:
It might not be needed. It creates a zip even when the second iteratable is emtpy.